docs/4.3/openssh_teleport.md

```diff
@@ -0,0 +1,311 @@
+## Using OpenSSH Client
+
+It is possible to use the OpenSSH client `ssh` to connect to nodes within a
+Teleport cluster. Teleport supports SSH subsystems and includes a `proxy`
+subsystem that can be used like `netcat` is with `ProxyCommand` to connect
+through a jump host.
+
+On your client machine, you need to import these keys. It will allow your
+OpenSSH client to verify that host's certificates are signed by the trusted CA
+key:
+
+``` yaml
+$ cat teleport_user_ca.pub >> ~/.ssh/known_hosts
+```
+
+If you have multiple Teleport clusters, you have to export and set up these
+certificate authorities for each cluster individually.
+
+!!! tip "OpenSSH and Trusted Clusters"
+
+    If you use [recording proxy mode](architecture/teleport_proxy.md) and [trusted
+    clusters](trusted-clusters.md), you need to set up certificate authority from
+    the _root_ cluster to match **all** nodes, even those that belong to _leaf_
+    clusters. For example, if your node naming scheme is `*.root.example.com`,
+    `*.leaf1.example.com`, `*.leaf2.example.com`, then the
+    `@certificate-authority` entry should match `*.example.com` and use the CA
+    from the root auth server only.
+
+Make sure you are running OpenSSH's `ssh-agent` , and have logged in to the
+Teleport proxy:
+
+``` bash
+$ eval `ssh-agent`
+$ tsh --proxy=work.example.com login
+```
+
+`ssh-agent` will print environment variables into the console. Either `eval` the
+output as in the example above, or copy and paste the output into the shell you
+will be using to connect to a Teleport node. The output exports the
+`SSH_AUTH_SOCK` and `SSH_AGENT_PID` environment variables that allow OpenSSH
+clients to find the SSH agent.
+
+Lastly, configure the OpenSSH client to use the Teleport proxy when connecting
+to nodes with matching names. Edit `~/.ssh/config` for your user or
+`/etc/ssh/ssh_config` for global changes:
+
+``` bash
+# work.example.com is the jump host (proxy). credentials will be obtained from the
+# openssh agent.
+Host work.example.com
+    HostName 192.168.1.2
+    Port 3023
+
+# connect to nodes in the work.example.com cluster through the jump
+# host (proxy) using the same. credentials will be obtained from the
+# openssh agent.
+Host *.work.example.com
+    HostName %h
+    Port 3022
+    ProxyCommand ssh -p 3023 %r@work.example.com -s proxy:%h:%p
+
+# when connecting to a node within a trusted cluster with name "remote-cluster",
+# add the name of the cluster to the invocation of the proxy subsystem.
+Host *.remote-cluster.example.com
+   HostName %h
+   Port 3022
+   ProxyCommand ssh -p 3023 %r@work.example.com -s proxy:%h:%p@remote-cluster
+```
+
+When everything is configured properly, you can use ssh to connect to any node
+behind `work.example.com` :
+
+``` bash
+$ ssh root@database.work.example.com
+```
+
+!!! tip "Note"
+
+    Teleport uses OpenSSH certificates instead of keys which means
+    you can not connect to a Teleport node by IP address. You have to connect by
```

Comment:
```suggestion
    you cannot ordinarily connect to a Teleport node by IP address. You have to connect by
```

I think you can actually allow this by putting the node's IP into `advertise_ip` or `ssh_service.public.addr` and restarting Teleport (as it'll regenerate the node's cert to include the IP as a principal) but this might be advanced usage. Up to you whether we highlight it or not.

Reply:
I was chatting to @russjones about this, this makes a lot more sense now why using IP sucks. I'll make a note of this.